### PR TITLE
Issue #4647 Hazelcast remote.xml configuration file do not configure hazelcast remote addresses 

### PR DESCRIPTION
**Pull request by attiand**
Signed-off-by: Mattias Andersson <mattias.andersson@umu.se>

### CODE REVIEW

File: jetty-hazelcast/src/main/config/modules/session-store-hazelcast-remote.mod [hunk 1]
```diff
@@ -34,6 +34,6 @@ jetty.session.hazelcast.mapName=jetty-distributed-session-map
 jetty.session.hazelcast.hazelcastInstanceName=JETTY_DISTRIBUTED_SESSION_INSTANCE
 jetty.session.hazelcast.onlyClient=true
 jetty.session.hazelcast.scavengeZombies=false
-#jetty.session.hazelcast.configurationLocation=
 jetty.session.gracePeriod.seconds=3600
 jetty.session.savePeriod.seconds=0
+#jetty.session.hazelcast.addresses=192.168.1.5,192.168.1.6
```
Comment (olamy) [inline jetty-hazelcast/src/main/config/modules/session-store-hazelcast-remote.mod]: please remove those ips

**Review by olamy — CHANGES_REQUESTED** [review]
Please sign the eclipse contributor agreement  https://www.eclipse.org/legal/ECA.php
And signed your commit with the same email

File: jetty-hazelcast/src/main/java/org/eclipse/jetty/hazelcast/session/HazelcastSessionDataStoreFactory.java [hunk 2]
```diff
@@ -81,6 +85,12 @@ public SessionDataStore getSessionDataStore(SessionHandler handler)
                     if (configurationLocation == null)
                     {
                         ClientConfig config = new ClientConfig();
+                        
+                        if (addresses != null && !addresses.isEmpty())
```
Comment (olamy) [inline jetty-hazelcast/src/main/java/org/eclipse/jetty/hazelcast/session/HazelcastSessionDataStoreFactory.java:89]: checkstyle will fail

File: jetty-hazelcast/src/main/java/org/eclipse/jetty/hazelcast/session/HazelcastSessionDataStoreFactory.java [hunk 3]
```diff
@@ -201,4 +211,14 @@ public void setHazelcastInstanceName(String hazelcastInstanceName)
     {
         this.hazelcastInstanceName = hazelcastInstanceName;
     }
+
```
Comment (olamy) [inline jetty-hazelcast/src/main/java/org/eclipse/jetty/hazelcast/session/HazelcastSessionDataStoreFactory.java:214]: checkstyle will fail

File: jetty-hazelcast/src/main/config/modules/session-store-hazelcast-embedded.mod [hunk 4]
```diff
@@ -31,7 +31,6 @@ http://www.apache.org/licenses/LICENSE-2.0.html
 [ini-template]
 jetty.session.hazelcast.mapName=jetty-distributed-session-map
 jetty.session.hazelcast.hazelcastInstanceName=JETTY_DISTRIBUTED_SESSION_INSTANCE
-#jetty.session.hazelcast.configurationLocation=
```
Comment (olamy) [inline jetty-hazelcast/src/main/config/modules/session-store-hazelcast-embedded.mod:34]: it must be still possible to configure hazelcast in embedded mode so I would prefer leave it but it's not in the xml and it should be maybe you can add it as well?

File: jetty-hazelcast/src/main/config/modules/session-store-hazelcast-remote.mod [hunk 5]
```diff
@@ -34,6 +34,6 @@ jetty.session.hazelcast.mapName=jetty-distributed-session-map
 jetty.session.hazelcast.hazelcastInstanceName=JETTY_DISTRIBUTED_SESSION_INSTANCE
 jetty.session.hazelcast.onlyClient=true
 jetty.session.hazelcast.scavengeZombies=false
-#jetty.session.hazelcast.configurationLocation=
```
Comment (olamy) [inline jetty-hazelcast/src/main/config/modules/session-store-hazelcast-remote.mod:37]: it must be still possible to configure hazelcast so I would prefer leave it but it's not in the xml and it should be maybe you can add it as well?

**Review by olamy — CHANGES_REQUESTED** [review]
thanks a lot but still a small change :)

**Comment by olamy** [comment]
@attiand forget my request regarding `configurationLocation` I will do it with another pr

**Comment by attiand** [comment]
ok @olamy, if the default value of the property is the empty string you need to add the check

```java
if (configurationLocation == null || configurationLocation.isEmpty())
```

**Pull request merged**